bigbluebutton-html5/imports/api/2.0/users/server/handlers/presenterAssigned.js

```diff
@@ -42,17 +43,18 @@ export default function handlePresenterAssigned({ body }, meetingId) {
     $set: {
       'user.presenter': true,
     },
+    $push: {
+      'user.roles': 'presenter',
+    },
   };
 
-  const cb = (err, numChanged) => {
+  const cb = (err) => {
     if (err) {
       return Logger.error(`Assigning user as presenter: ${err}`);
     }
 
-    if (numChanged) {
-      unassignCurrentPresenter(meetingId, presenterId);
-      return Logger.info(`Assigned user as presenter id=${presenterId} meeting=${meetingId}`);
-    }
+    unassignCurrentPresenter(meetingId, presenterId);
```

Comment:
You need to check the `numChanged` otherwise in some odd cases you end up without any presenter.